tasks/tools.py

```diff
@@ -49,12 +53,113 @@ def remove_from_do(id):
 
     session.execute("COMMIT")
 
+
+def extract_nulls_from_log(infile, outfile, relevant_fields):
+    STR_NULL = 'contain only NULL values: '
+    STR_ERROR = '[ERROR]'
+    STR_INFO = '[INFO]'
+    STR_EXCEPTIONS = 'exceptions'
+
+    relevant_fields = relevant_fields.split(',')
+    cols = {}
+
+    try:
+        os.remove(outfile)
+    except OSError:
+        pass
+
+    with open(outfile, 'a') as fout:
+        with open(infile) as fin:
+            current_params = {}
+            for line in fin:
+                if STR_ERROR in line:
+                    current_params = {}
+                    try:
+                        vals = {item.split("=")[0].strip(): item.split("=")[1].strip()
+                                for item in line[line.rfind('(') + 1:line.rfind(')')].split(',')}
+                        for key, val in vals.items():
+                            if key in relevant_fields:
+                                current_params[key] = val
+                    except Exception as e:
+                        pass
+
+                elif STR_NULL in line and STR_INFO not in line:
+                    columns = [x.strip() for x in line[line.rfind(STR_NULL) + len(STR_NULL):].split(',')]
+
+                    for column in columns:
+                        if column in cols:
+                            if current_params not in cols[column][STR_EXCEPTIONS]:
+                                cols[column][STR_EXCEPTIONS].extend([current_params])
```

Comment:
`extend` with an array with a single equivalent should be equivalent to `append`:

```suggestion
                                cols[column][STR_EXCEPTIONS].append(current_params)
```

?